### PR TITLE
✨ Include version in origin field for debian packages

### PR DESCRIPTION
**Pull request by czunker**
Some packages contain the version information for the source package. Where we can find it, we add it.

### CODE REVIEW

**Comment by czunker** [comment]
An exmaple:
```
cnquery run container image samba:bookworm -c "packages.where(origin == /samba/){ name version origin originVersion }" 
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
packages.where.list: [
  0: {
    name: "libldb2"
    version: "2:2.6.2+samba4.17.12+dfsg-0+deb12u1"
    originVersion: "2:4.17.12+dfsg-0+deb12u1"
    origin: "samba"
  }
  1: {
    name: "python3-ldb"
    version: "2:2.6.2+samba4.17.12+dfsg-0+deb12u1"
    originVersion: "2:4.17.12+dfsg-0+deb12u1"
    origin: "samba"
  }
]
```

File: providers/os/resources/os.lr [hunk 1]
```diff
@@ -616,6 +616,9 @@ package @defaults("name version") {
 
   // Package origin (optional)
   origin() string
+  // Package origin version (optional)
+  originVersion() string
```
Comment (chris-rock) [inline providers/os/resources/os.lr]: should this not be origin package that references another full package? We have the following options:

```
// dict returns name and version field
originPackage() dict

// return full package resource
originPackage() package
```

**Review by chris-rock — COMMENTED** [review]
I added some comment about the MQL resource. Do we also need to adjust other operating systems that have origin support or is that just debian?

**Comment by czunker** [comment]
> I added some comment about the MQL resource. Do we also need to adjust other operating systems that have origin support or is that just debian?

I checked Alpine. That does not provide a version, only the origin name.

File: providers/os/resources/os.lr (hunk 1 above)
Comment (czunker) [reply providers/os/resources/os.lr]: As discuessed, I now include the version in the origin field as it is shown in dpkg output:
```
samba (2:4.17.12+dfsg-0+deb12u1)
```

File: providers/os/resources/packages.go [hunk 2]
```diff
@@ -5,10 +5,11 @@ package resources
 
 import (
 	"errors"
-	"go.mondoo.com/cnquery/v10/types"
 	"regexp"
 	"sync"
 
+	"go.mondoo.com/cnquery/v10/types"
+
```
Comment (chris-rock) [inline providers/os/resources/packages.go]: nit: whitespace

File: providers/os/resources/os.lr [hunk 3]
```diff
@@ -614,7 +614,7 @@ package @defaults("name version") {
   // Common Platform Enumeration (CPE) for the package
   cpes []core.cpe
 
-  // Package origin (optional)
+  // Package origin (optional) (This includes the version when present)
```
Comment (chris-rock) [inline providers/os/resources/os.lr]:
```suggestion
  // Package origin, may includes version if available (optional)
```

File: providers/os/resources/packages/dpkg_packages.go [hunk 4]
```diff
@@ -6,15 +6,16 @@ package packages
...
 	"io"
 	"os"
 	"regexp"
 	"strings"
 
+	"github.com/package-url/packageurl-go"
+	"go.mondoo.com/cnquery/v10/providers-sdk/v1/inventory"
+	"go.mondoo.com/cnquery/v10/providers/os/resources/cpe"
+	"go.mondoo.com/cnquery/v10/providers/os/resources/purl"
+
```
Comment (chris-rock) [inline providers/os/resources/packages/dpkg_packages.go]: nit: whitespace

**Review by chris-rock — APPROVED** [review]
Thank you @czunker

**Pull request merged**